### PR TITLE
Changing RowNumberPaging should cause new internal service provider to be built

### PR DESCRIPTION
**Pull request by ajcvickers**
Issue #8877

### CODE REVIEW

File: src/EFCore.SqlServer/Infrastructure/Internal/SqlServerOptionsExtension.cs [hunk 1]
```diff
@@ -61,6 +62,16 @@ public virtual SqlServerOptionsExtension WithRowNumberPaging(bool rowNumberPagin
             return clone;
         }
 
+        public override long GetServiceProviderHashCode()
+        {
+            if (_serviceProviderHash == null)
+            {
+                _serviceProviderHash = (base.GetServiceProviderHashCode() * 397) ^ _rowNumberPaging.GetHashCode();
```
Comment (smitpatel) [inline src/EFCore.SqlServer/Infrastructure/Internal/SqlServerOptionsExtension.cs]: null check on `_rowNumberPaging` ?

File: test/EFCore.SqlServer.Tests/SqlServerOptionsExtensionTest.cs [hunk 2]
```diff
@@ -20,5 +23,81 @@ public void ApplyServices_adds_SQL_server_services()
...
+                                if (_rowNumberPagingEnabled)
+                                {
+                                    b.UseRowNumberForPaging();
+                                }
+                            });
+            }
+        }
     }
+
+
```
Comment (smitpatel) [inline test/EFCore.SqlServer.Tests/SqlServerOptionsExtensionTest.cs]: extra lines

Comment (ajcvickers) [inline test/EFCore.SqlServer.Tests/SqlServerOptionsExtensionTest.cs]: Fixed

File: src/EFCore.SqlServer/Infrastructure/Internal/SqlServerOptionsExtension.cs (hunk 1 above)
Comment (ajcvickers) [inline src/EFCore.SqlServer/Infrastructure/Internal/SqlServerOptionsExtension.cs]: Fixed

Comment (caleblloyd) [inline src/EFCore.SqlServer/Infrastructure/Internal/SqlServerOptionsExtension.cs]: Just out of curiosity, why 397? Just any old prime number?

Comment (smitpatel) [inline src/EFCore.SqlServer/Infrastructure/Internal/SqlServerOptionsExtension.cs]: https://stackoverflow.com/questions/102742/why-is-397-used-for-resharper-gethashcode-override

**Pull request merged**